ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -808,4 +830,58 @@ object Serializers {
       strings(input.readInt())
     }
   }
+
+  private class RewriteArgumentsTransformer extends Transformers.Transformer {
+    import RewriteArgumentsTransformer._
+
+    private[this] var paramToIndex: Map[String, Int] = _
+
+    def transformMethodDef(tree: MethodDef): MethodDef = {
+      /* Ideally, we would re-hash the new MethodDef here, but we cannot do
+       * that because it prevents the JS version of the tools to link.
+       * Since the hashes of exported methods are not used by our pipeline
+       * anyway, we simply put None.
+       */
+      val MethodDef(static, name, args, resultType, body) = tree
+      setupParamToIndex(args)
+      MethodDef(static, name, List(argumentsParamDef(tree.pos)),
```

Comment:
We should probably fast fail here, if this method def is not exported... Otherwise we'll produce invalid IR IIUC.